docs/dev/metrics.md

```diff
@@ -0,0 +1,190 @@
+# MAO Metrics
+
+The Machine API Operator reports the following metrics:
+
+Metrics about Machine resources managed by the operator:
+
+```
+# HELP mapi_machine_created_timestamp_seconds Timestamp of the mapi managed Machine creation time
+# TYPE mapi_machine_created_timestamp_seconds gauge
+mapi_machine_created_timestamp_seconds{api_version="machine.openshift.io/v1beta1",name="ocp-cluster-rndpg-master-0",namespace="openshift-machine-api",node="ip-10-0-130-139.us-east-2.compute.internal",phase="Running",spec_provider_id="aws:///us-east-2a/i-08624d119917119d6"} 1.589550152e+09
```

Comment:
I notice there are some labels on this that might change through the lifecycle of a machine, eg phase, does this really just report when a machine was created? Or is there something deeper on this that's not being expressed by the name? I wonder if this needs a bit more explanation

Reply:
that's a good question. i produced this output by scraping the mao on a running cluster i had been using for testing, i will look into this a little deeper.

Reply:
ok, did some checking. the`mapi_machine_created_timestamp_seconds` and `mapi_machine_set_created_timestamp_seconds` will both update their `.status.phase` on every update cycle (not sure on the frequency here). theoretically, the other values could change as well but it seems like `phase` is the only one that will change.

should i add a note about this behavior?

Reply:
I'd be tempted to add some detail if you think this will add understanding for the metrics. 

At the moment, this list to me is just a list of prometheus metrics, it's still quite hard to interpret. If there's any detail we can work out and add to explain what the metrics are/why they exist, that's helpful for future readers so they don't also have to try and interpret the metrics scrape too

Reply:
i think that's fair. my thought process here was to put up something with the raw scrape, broken into sections, to help start the ball rolling. i'm a little torn about how much detail to add here, but i'll go back and give it another pass.